includes/class-wc-query.php

```diff
@@ -598,10 +598,14 @@ public function order_by_popularity_post_clauses( $args ) {
 	 * @return array
 	 */
 	public function get_meta_query( $meta_query = array(), $main_query = false ) {
+		global $wp_query;
+
 		if ( ! is_array( $meta_query ) ) {
 			$meta_query = array();
 		}
-		$meta_query['price_filter'] = $this->price_filter_meta_query();
+		if ( $main_query || $wp_query === self::$product_query ) {
```

Comment:
This comparison is not needed - $main_query is passed. The functions calling this method take care of main query checks.

Reply:
Thanks @mikejolley . WP query check removed.

Reply:
Don't forget the global :)

Reply:
Thanks :) Removed.